addon/components/high-charts.js

```diff
@@ -38,40 +38,32 @@ export default Component.extend({
   didReceiveAttrs() {
     this._super(...arguments);
 
-    if (!(get(this, 'content') && get(this, 'chart'))) {
-      return;
-    }
+    const content = get(this, 'content');
+    const chart = get(this, 'chart');
+    if (!content || !chart) return;
 
-    let chart = get(this, 'chart');
     let noData = chart.get('noData');
+    if (noData != null) noData.remove();
 
-    if (noData != null) {
-      noData.remove();
-    }
-
-    let numToRemove = chart.series.length - get(this, 'content').length;
-
-    for (let i = numToRemove; i > 0; i--) {
-
-      let lastIndex = chart.series.length - 1;
+    // remove series that aren't defined any more
+    // udpate series with new data
+    chart.series.forEach((series) => {
+      const contentSeries = content.filter((item) => item.name === series.name);
+      if (series.name === 'Navigator') return true;
```

Comment:
From what I can tell, it seems like after the initial render highcharts adds an additional navigator series if the navigator is enabled. That was throwing off the original code where it was expecting `content` to match `chart.series`.